terraso_backend/apps/core/models/groups.py

```diff
@@ -110,6 +127,8 @@ class Membership(BaseModel):
 
     user_role = models.CharField(max_length=64, choices=ROLES, blank=True, default=ROLE_MEMBER)
 
+    objects = MembershipManager()
```

Comment:
> By default, Django adds a Manager with the name objects to every Django model class.

[ref](https://docs.djangoproject.com/en/4.0/topics/db/managers/#manager-names)

Reply:
Question: What is the benefit of this? can we add the `managers_only` function directly in the group model?

Reply:
Yes, we could add directly to the model. But we shouldn't, because the manager is the place to do this kind of work.

As per the documentation:
> A Manager is the interface through which database query operations are provided to Django models.

Following the standard, we have benefits. For example, having the functions available on every place where Django uses the model manager. The change on this PR is a good example. The `managers_only()` function was added to the `Membership` model manager. As a nice consequence, it can be used on backward relationship access. More [here](https://docs.djangoproject.com/en/4.0/topics/db/queries/#using-a-custom-reverse-manager). 

The way we use the function added to the `Membership` model manager. Look that we access it through a reverse manager called `memberships`.
```python
# On User model
def is_group_manager(self, group_id):
    return self.memberships.managers_only().filter(group__pk=group_id).exists()
````

Reply:
Ohh, I see, so this `Manager` definition has nothing to do with the `manager` definition from the group roles perspective, right?

Reply:
Yes :+1:

Reply:
@cacarrara in that case can we rename it to avoid this confusion, it seems we are closely using the same terminology for 2 separate concepts

Reply:
Maybe `MembershipOperations` to try to keep the django meaning

Reply:
For now, I would rather keep as it is. I think the conceptual and place distance between the two "manager" things here won't be a problem. On the other hand, renaming the well-known Django's Manager component might lead to misunderstandings. Specially because on other models the "manager" name would still be used. We already make use of it on `User` model, for example. Of course, we can reevaluate it in the future.

Reply:
Issue: That doesn't seem to be the case looking at the code:
```
class MembershipManager(models.Manager):
    def managers_only(self):
        return self.filter(user_role=Membership.ROLE_MANAGER)
```
In this piece of code just involving 3 lines I see "manager" being referenced 4 times with different meanings

Reply:
@josebui before further messages, is it really blocking?

Reply:
@cacarrara I think so, from the readability point of view, this is really confusing IMO, because you are reading code with product knowledge, there is a cognitive load trying to identify what `manager`s are referencing django and which ones are referencing the definition from the product side. So, to negotiate this I see these options:
- We rename it to avoid the confusion
- We create another issue with the confusion stated in the problem with the goal to indentify a different name the Managers from Django should use
- Have a call for you to  explain further why changing this name is not a good idea and why this is not confusing reading the code
- Paul think this is fine, and there is no confusion reading the code, getting a majority to support this approach

Reply:
@josebui I renamed it. Let me know what you think.

Reply:
@cacarrara That makes sense to me, still a little confusing, but the `MembershipObjectsManager` makes it clearer that this is managing objects and that added some distance with the Manager product definition, I am ok with this